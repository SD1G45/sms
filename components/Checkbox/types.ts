import React from "react";

export interface CheckboxProps extends React.HTMLAttributes<HTMLInputElement> {
  checked: boolean;
}
