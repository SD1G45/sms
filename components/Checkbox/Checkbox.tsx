import React from "react";
import styled from "styled-components";
import {
  CheckboxContainer,
  HiddenCheckbox,
  Icon,
  StyledCheckbox,
} from "./styles";

const Checkbox: React.FC<any> = ({ label, className, checked, ...other }) => (
  <label>
    <CheckboxContainer className={className}>
      <HiddenCheckbox checked={checked} {...other} />
      <StyledCheckbox checked={checked}>
        <Icon viewBox="0 0 24 24">
          <polyline points="20 6 9 17 4 12" />
        </Icon>
      </StyledCheckbox>
    </CheckboxContainer>
    <span style={{ marginLeft: 8 }}>{label}</span>
  </label>
);

export default Checkbox;
