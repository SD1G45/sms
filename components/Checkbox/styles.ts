import styled from "styled-components";

export const CheckboxContainer = styled.div<any>`
  display: inline-block;
  vertical-align: middle;
`;

export const Icon = styled.svg<any>`
  fill: none;
  stroke: white;
  stroke-width: 2px;
`;
// Hide checkbox visually but remain accessible to screen readers.
// Source: https://polished.js.org/docs/#hidevisually
export const HiddenCheckbox = styled.input.attrs({ type: "checkbox" })`
  border: 0;
  clip: rect(0 0 0 0);
  clippath: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`;

export const StyledCheckbox = styled.div<any>`
  display: inline-block;
  width: 16px;
  height: 16px;
  background: ${(props) =>
    props.checked ? props.theme.colors.primary : "#9ED7F6"};
  border-radius: 3px;
  transition: all 150ms;

  ${HiddenCheckbox}:focus + & {
    box-shadow: 0 0 0 3px ${(props) => props.theme.colors.primary};
  }

  ${Icon} {
    visibility: ${(props) => (props.checked ? "visible" : "hidden")};
  }
`;
