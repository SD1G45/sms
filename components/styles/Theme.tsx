import React from "react";
import { ThemeProvider } from "styled-components";
import ITheme from "./types";

const screenWidth = {
  minSmall: `(min-width: ${430}px)`,
  minMedium: `(min-width: ${730}px)`,
  minLarge: `(min-width: ${1024}px)`,
  maxSmall: `(max-width: ${430}px)`,
  maxMedium: `(max-width: ${730}px)`,
  maxLarge: `(max-width: ${1024}px)`,
};

const lightTheme: ITheme = {
  colors: {
    primary: "#4881F0",
    lightGray: "#BEC8D2",
    textPrimary: "#000000",
    error: "#FF7F7F",
  },
  screenWidth: { ...screenWidth },
};

interface ThemeProps {
  children: React.ReactNode;
}

const Theme: React.FC<ThemeProps> = ({ children }) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);

export default Theme;
