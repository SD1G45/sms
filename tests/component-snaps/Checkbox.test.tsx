import React from 'react';
import renderer from "react-test-renderer";
import "jest-styled-components";
import Checkbox from '../../components/Checkbox';
import Theme from "../../components/styles/Theme";

test("Checkbox renders correctly unchecked", () => {
  const tree = renderer
    .create(
      <Theme>
        <Checkbox checked={false}/>
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Checkbox renders correctly checked", () => {
  const tree = renderer
    .create(
      <Theme>
        <Checkbox checked={true}/>
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});