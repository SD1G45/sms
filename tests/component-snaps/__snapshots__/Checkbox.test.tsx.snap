// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly checked 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
}

.c6 {
  fill: none;
  stroke: white;
  stroke-width: 2px;
}

.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clippath: inset(50%);
  clippath: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  display: inline-block;
  width: 16px;
  height: 16px;
  background: #4881F0;
  border-radius: 3px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
}

.c1:focus + .c3 {
  box-shadow: 0 0 0 3px #4881F0;
}

.c4 .c5 {
  visibility: visible;
}

<label>
  <div
    className="c0"
  >
    <input
      checked={true}
      className="c1 c2"
      type="checkbox"
    />
    <div
      checked={true}
      className="c3 c4"
    >
      <svg
        className="c5 c6"
        viewBox="0 0 24 24"
      >
        <polyline
          points="20 6 9 17 4 12"
        />
      </svg>
    </div>
  </div>
  <span
    style={
      Object {
        "marginLeft": 8,
      }
    }
  />
</label>
`;

exports[`Checkbox renders correctly unchecked 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
}

.c6 {
  fill: none;
  stroke: white;
  stroke-width: 2px;
}

.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clippath: inset(50%);
  clippath: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c4 {
  display: inline-block;
  width: 16px;
  height: 16px;
  background: #9ED7F6;
  border-radius: 3px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
}

.c1:focus + .c3 {
  box-shadow: 0 0 0 3px #4881F0;
}

.c4 .c5 {
  visibility: hidden;
}

<label>
  <div
    className="c0"
  >
    <input
      checked={false}
      className="c1 c2"
      type="checkbox"
    />
    <div
      checked={false}
      className="c3 c4"
    >
      <svg
        className="c5 c6"
        viewBox="0 0 24 24"
      >
        <polyline
          points="20 6 9 17 4 12"
        />
      </svg>
    </div>
  </div>
  <span
    style={
      Object {
        "marginLeft": 8,
      }
    }
  />
</label>
`;
